components/core/src/crypto/keys/mod.rs

```diff
@@ -578,18 +601,33 @@ mod test {
         // we need to wait at least 1 second between generating keypairs to ensure uniqueness
         thread::sleep(time::Duration::from_millis(1000));
         let _ = SigKeyPair::generate_pair_for_origin("foo", cache.path());
-        let revs = super::get_key_revisions("foo", cache.path(), None).unwrap();
+        let revs = super::get_key_revisions("foo", cache.path(), None, &KeyType::Sig).unwrap();
         assert_eq!(2, revs.len());
     }
 
     #[test]
+    fn get_key_revisions_can_only_return_keys_of_specified_type() {
+        let cache = TempDir::new("key_cache").unwrap();
+        let _ = SigKeyPair::generate_pair_for_origin("foo", cache.path());
+        let revs = super::get_key_revisions("foo", cache.path(), None, &KeyType::Sig).unwrap();
+        assert_eq!(1, revs.len());
+        // we need to wait at least 1 second between generating keypairs to ensure uniqueness
```

Comment:
totally! would they be unique? would they be the same? It was like an eternity passed in that one second.